src/Documents/Session/IDocumentQuery.ts

```diff
@@ -1,50 +1,48 @@
-  or: IDocumentQuery<T>;
-  and: IDocumentQuery<T>;
-  not: IDocumentQuery<T>;
+  first(callback?: EntityCallback<T>): Promise<T>;
+  count(callback?: EntitiesCountCallback): Promise<number>;
+  take(docsCount: number);
+  skip(skipCount: number);
+  andAlso();
+  orElse();
+  negateNext();
   selectFields(...args: string[]): IDocumentQuery<T>;
-  search(fieldName: string, searchTerms: string | string[], escapeQueryOptions?: EscapeQueryOption, boost?: number): IDocumentQuery<T>;
-  where(conditions: IDocumentQueryConditions): IDocumentQuery<T>;
-  whereEquals<V extends LuceneValue>(fieldName: string, value: V, escapeQueryOptions?: EscapeQueryOption): IDocumentQuery<T>;
-  whereEndsWith(fieldName: string, value: string): IDocumentQuery<T>;
-  whereStartsWith(fieldName: string, value: string): IDocumentQuery<T>;
-  whereIn<V extends LuceneValue>(fieldName: string, values: V[]): IDocumentQuery<T>;
-  whereBetween<V extends LuceneValue>(fieldName: string, start?: V, end?: V): IDocumentQuery<T>;
-  whereBetweenOrEqual<V extends LuceneValue>(fieldName: string, start?: V, end?: V): IDocumentQuery<T>;
-  whereGreaterThan<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereGreaterThanOrEqual<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereLessThan<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereLessThanOrEqual<V extends LuceneValue>(fieldName: string, value: V): IDocumentQuery<T>;
-  whereIsNull(fieldName: string): IDocumentQuery<T>;
-  whereNotNull(fieldName: string): IDocumentQuery<T>;
-  orderBy(fieldsNames: string | string[]): IDocumentQuery<T>;
-  orderByDescending(fieldsNames: string | string[]): IDocumentQuery<T>;
-  andAlso(): IDocumentQuery<T>;
-  orElse(): IDocumentQuery<T>;
-  negateNext(): IDocumentQuery<T>;
-  take(docsCount: number): IDocumentQuery<T>;
-  skip(skipCount: number): IDocumentQuery<T>;
+  search(from: string, searchTerms: string | string[], boostField, boostValue, boostExpression, count): IDocumentQuery<T>;
+  where(conditions: IDocumentQueryConditions);
+  whereEquals<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  endsWith<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  exact<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  startsWith<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  whereIn<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  whereBetween<V extends RQLValue>(field: string, start?: V, end?: V, orName?, orValue?): IDocumentQuery<T>;
+  whereGreaterThan<V extends RQLValue>(field: string, value: V): IDocumentQuery<T>;
+  whereGreaterThanOrEqual<V extends RQLValue>(field: string, value: V, orName?, orValue?): IDocumentQuery<T>;
+  whereEqualsAndOr<V extends RQLValue>(fieldFirst: string, valueFirst: string, fieldSecond: string, valueSecond: number, fieldThird: string, valueThird: number): IDocumentQuery<T>;
```

Comment:
There's no such query method in C# client. We'd like Client API to be consistent across platforms. Only query methods listed in https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Session/IDocumentQueryBase.cs
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Session/IDocumentQuery.cs

should be available.